app/models/manager_refresh/save_collection/saver/base.rb

```diff
@@ -37,21 +37,30 @@ def initialize(inventory_collection)
                                                 .try(:[], "sql_type")
         end
 
-        @serializable_keys = @model_class.attribute_names.each_with_object({}) do |key, obj|
+        @serializable_keys = {}
+        @deserializable_keys = {}
+        @model_class.attribute_names.each do |key|
           attribute_type = @model_class.type_for_attribute(key.to_s)
           pg_type        = @pg_types[key.to_sym]
 
           if inventory_collection.use_ar_object?
             # When using AR object, lets make sure we type.serialize(value) every value, so we have a slow but always
             # working way driven by a configuration
-            obj[key.to_sym] = attribute_type
+            @serializable_keys[key.to_sym] = attribute_type
+            @deserializable_keys[key.to_sym] = attribute_type
```

Comment:
Initially I thought this unnecessary, but then I wrote the test and it failed with {:saver_strategy=>:batch, :use_ar_object=>**true**} 
(duh, that's the one container refresh tests don't exercise :wink:).
Turns out `map_ids_to_inventory_objects` ignores use_ar_object and takes values from DB directly anyway (unless it has to use fallback query mode, with unrelated conditions).
So I set `deserializable_keys` in use_ar_object mode too.
Should I rewrite `map_ids_to_inventory_objects` to check `use_ar_object`?

Reply:
```Should I rewrite map_ids_to_inventory_objects to check use_ar_object?```

Yeah, sound like we have to do it. And we should add 'use_ar_object' to containers specs.